docs/topics/control-flow.md

```diff
@@ -177,6 +177,82 @@ fun Request.getBody() =
 
 The scope of variable introduced in *when* subject is restricted to the body of this *when*.
 
+### Guard conditions in when expressions
+
+> Guard conditions are an [experimental feature](components-stability.md#stability-levels-explained) that may be changed at any time.
+> We would appreciate your feedback in [YouTrack](https://youtrack.jetbrains.com/issue/KT-71140/Guard-conditions-in-when-expressions-feedback).
+>
+{type="warning"}
+
+Starting from Kotlin 2.1, you can use guard conditions in `when` expressions or statements with a subject.
+
+Guard conditions allow you to include more than one condition to the branches of a `when` expression, making complex control flow more explicit and concise.
+
+To include a guard condition in a branch, place the guard condition together with the `if` keyword after the primary condition:
+
+```kotlin
+sealed interface Animal {
+    class Cat(val mouseHunter: Boolean) : Animal
+    data object Dog : Animal
```

Comment:
This seems a bit weird, as `object` seems to imply that there's just one single dog.

```suggestion
    data class Dog(val breed: String) : Animal
```